testing/x/replay.go

```diff
@@ -31,6 +31,10 @@ import (
 	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 )
 
+type ReplayOptions struct {
+	MatchOptions JSONMatchOptions
```

Comment:
Are these options always for "failures" in response? I think we can just assume this as part of Replay. It's fair game for Replay to know about Pulumi protocol semantics. If we don't care about failure ordering we can teach Replay to always match "failures" as sets. Then this slightly clunky option goes away.

Reply:
That sounds like the best thing to do - I avoided it thinking it'd be too specific for the library but I think you have a good point that replay already knows quite a bit about the protocol!

I'll just fix the ordering issue there and remove all the stuff with the options.